otherlibs/stdune/src/option.mli

```diff
@@ -60,3 +60,21 @@ module List : sig
 end
 
 val merge : 'a t -> 'a t -> f:('a -> 'a -> 'a) -> 'a t
+
+module Unboxed : sig
+  (** Poor man's unboxed option types. The value stored must not be immediate *)
```

Comment:
It may not be obvious from this comment that you can't use `'a t t`, i.e. you can't nest such cheap unboxed options.

Reply:
Add a comment regarding this use case.